core/src/commonMain/kotlin/com/ustadmobile/core/networkmanager/ImportJobRunner.kt

```diff
@@ -0,0 +1,173 @@
+package com.ustadmobile.core.networkmanager
+
+import com.ustadmobile.core.account.Endpoint
+import com.ustadmobile.core.contentformats.ContentImportManager
+import com.ustadmobile.core.db.JobStatus
+import com.ustadmobile.core.db.UmAppDatabase
+import com.ustadmobile.core.networkmanager.downloadmanager.ContainerDownloadManager
+import com.ustadmobile.core.util.UMFileUtil
+import com.ustadmobile.door.DoorLiveData
+import com.ustadmobile.door.DoorMutableLiveData
+import com.ustadmobile.lib.db.entities.ConnectivityStatus
+import com.ustadmobile.lib.db.entities.ContainerImportJob
+import com.ustadmobile.lib.db.entities.ContainerWithContainerEntryWithMd5
+import io.ktor.client.request.header
+import io.ktor.client.request.parameter
+import io.ktor.client.request.post
+import io.ktor.client.request.url
+import io.ktor.client.statement.HttpStatement
+import io.ktor.http.HttpStatusCode
+import kotlinx.atomicfu.atomic
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import org.kodein.di.DI
+import org.kodein.di.DIAware
+import org.kodein.di.instance
+import org.kodein.di.on
+
+class ImportJobRunner(private val containerImportJob: ContainerImportJob, private val retryDelay: Long = DEFAULT_RETRY_DELAY, private val endpointUrl: String, override val di: DI) : DIAware {
+
+    private val db: UmAppDatabase by di.on(Endpoint(endpointUrl)).instance(tag = UmAppDatabase.TAG_DB)
+
+    private val containerUploader: ContainerUploaderCommon by instance()
+
+    private val contentImportManager: ContentImportManager by on(Endpoint(endpointUrl)).instance()
+
+    private val containerManager: ContainerDownloadManager by on(Endpoint(endpointUrl)).instance()
+
+    private val currentUploadAttempt = atomic(null as Deferred<Int>?)
+
+    private val currentHttpClient = defaultHttpClient()
+
+    private val _connectivityStatus = DoorMutableLiveData<ConnectivityStatus>()
+
+    val connectivityStatus: DoorLiveData<ConnectivityStatus>
+        get() = _connectivityStatus
+
+    /**
+     * Lock used for any operation that is changing the status of the Fetch download
+     */
+    private val downloadStatusLock = Mutex()
+
+    suspend fun importContainer(isFileImport: Boolean = false) {
```

Comment:
I think the isFileImport name is misleading. It is always a file import. Maybe rename the parameter to markContainerAsDownloaded ?

Reply:
done